synapse/dendrite/jobs.py

```diff
@@ -0,0 +1,214 @@
+import json
+import time
+import logging
+import collections
+import synapse.common as s_common
+import synapse.cortex as s_cortex
+import synapse.eventbus as s_eventbus
+import synapse.telepath as s_telepath
+
+logger = logging.getLogger(__name__)
+
+class Jobs(s_eventbus.EventBus):
+    '''
+    Jobs provides a simple remote queue for managing Parser jobs in Dendrite.
+
+    Parser jobs are added to named queues, an event is fired on the Service Bus, the Parsers registered for
+    the named queue receive the event and proceed to retrieve the job for processing. Basic job lifecycle management
+    is provided where a job can be in one of the following states: queued, working, and completed.
+
```

Comment:
In short, yes. My philosophy on this has been to defer to deployment/orchestration for managing how many workers are running and on what queue they are listening to. 300 workers racing to a small queue would definitely be inefficient. However, I can envision a system where as a queue's size trends upwards, an orchestration layer would ramp up other parsers to process the jobs, then ramp back down appropriately.